### PR TITLE
Enhancing `AlphaStack` component and changing to stable component

### LINKED ISSUE #1835: Enhancement `AlphaStack` layout component
- Fixes #1673

### PR DESCRIPTION
**Pull request by sungik-choi**
## Self Checklist

- [x] I wrote a PR title in **English** and added an appropriate **label** to the PR.
- [x] I wrote the commit message in **English** and to follow [**the Conventional Commits specification**](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I [added the **changeset**](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) about the changes that needed to be released. (or didn't have to)
- [x] I wrote or updated **documentation** related to the changes. (or didn't have to)
- [x] I wrote or updated **tests** related to the changes. (or didn't have to)
- [x] I tested the changes in various browsers. (or didn't have to)
  - Windows: Chrome, Edge, (Optional) Firefox
  - macOS: Chrome, Edge, Safari, (Optional) Firefox

## Related Issue

Fixes #1835

## Summary

(Changeset으로 갈음합니다)

`AlphaStack` component has been changed to a `Stack` component, and `Stack` component has been changed to `LegacyStack` component. Changes to remove the dependency of `Stack` and `StackItem` to allow stack layouts to be configured without additional DOM elements. And improved the existing `AlphaStack` to support more Flex layout related properties like reverse, wrap, and more align options, etc. See StackProps for more information.

We also added new `HStack` and `VStack` components, which are shorthand components that fix the direction prop of `AlphaStack`. As mentioned above, the existing components become `LegacyHStack`, `LegacyVStack`.

The layout implemented by `StackItem` can be partially replaced by the Layout component's `grow`, `shrink` common properties and margin common properties. Note that the `align`, `justify` (align-self, justify-self in CSS flex) properties provided by `StackItem` are not provided by the Layout component.

```jsx
/* AS-IS */
return (
  <Stack direction="horizontal" spacing={8} style={{ width: 300, height: 50 }}>
    <StackItem grow shrink weight={1} />
    <StackItem weight={0} size={10} marginBefore={20}>
      <Stack direction="vertical" />
    </StackItem>
  </Stack>
)

/* TO-BE */
return (
  <Stack direction="horizontal" spacing={8} width={300} height={50}>
    <Box grow={1} />
    <Stack direction="vertical" shrink={0} width={10} ml={12} />
  </Stack>
)
```

The changes also apply to other components that use `Stack` internally, and there are a few changes.

- `RadioGroup` component no longer supports `as` prop.
- `ButtonGroup` component now extends the interfaces of new `Stack`. It no longer supports `as` prop.
- `FormGroup` component now extends the interfaces of new `Stack`. It no longer supports `as` prop.

## Details

- AlphaStack을 Stack으로 변경합니다.
- Stack을 LegacyStack으로 변경합니다.
- 관련 타입 및 테스트를 마이그레이션합니다.

### Breaking change? (Yes/No)

Yes

## Referneces

- https://www.radix-ui.com/themes/docs/components/flex

### CODE REVIEW

File: packages/bezier-react/src/components/AlphaStack/AlphaStack.types.ts [hunk 1]
```diff
@@ -1,44 +1,58 @@
...
-   * Default spacing between children, in pixels.
-   *
-   * @remarks
-   * `spacing` could be enumerated in later design, if bezier design system
-   * decides to define a guideline for linear layout.
-   *
-   * @default 0
+   * Whether to reverse the order of children.
    */
-  spacing?: number
```
Comment (sungik-choi) [inline packages/bezier-react/src/components/AlphaStack/AlphaStack.types.ts:37]: spacing 토큰이 사라졌으며, 다음 디자인 시스템에서도 spacing을 enum으로 다루지 않을 것입니다. 좀 더 CSS에 가까운 이름인 `gap` 으로 속성명을 변경하고, string도 허용할 수 있게 변경했습니다. 똑같이 number type을 넘길 경우 암묵적으로 px unit을 붙이는 동작은 유지합니다.

Comment (sungik-choi) [reply packages/bezier-react/src/components/AlphaStack/AlphaStack.types.ts:37]: -> 생각 변경...

분명 `gap` 이 웹 표준과 일치하는 이름이며 간결하기에 좋긴 하나, 마이그레이션 코스트를 감안하여 이름을 `spacing` 으로 유지합니다.

**Comment by sungik-choi** [comment]
FYI. 빌드가 터지는 문제는 #1847 에서 해결합니다!

**Comment by sungik-choi** [comment]
FYI. CSS Custom property 상속 문제는 #1846 에서 해결합니다!

File: packages/bezier-react/src/components/Forms/FormGroup/FormGroup.types.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
-import type { LegacyStackProps } from '~/src/components/LegacyStack'
+import type { StackProps } from '~/src/components/Stack'
 
 interface FormGroupProps extends
-  BezierComponentProps,
+  AlphaBezierComponentProps,
   ChildrenProps,
-  Partial<Pick<LegacyStackProps, 'direction' | 'spacing'>>,
+  Partial<Pick<StackProps, 'direction' | 'spacing' | 'wrap'>>,
```
Comment (yangwooseong) [inline packages/bezier-react/src/components/Forms/FormGroup/FormGroup.types.ts:11]: 컴포넌트안에서 wrap prop 을 사용하지 않고 있어서 지정안해도 되지않을까요? 혹은 FormGroup.tsx L40에서 wrap={wrap} 으로 하는게 의도일까요?

File: .changeset/witty-pans-clean.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <StackItem weight={0} size={10} marginBefore={20}>
+      <Stack direction="vertical" />
+    </StackItem>
+  </Stack>
+)
+
+/* TO-BE */
+return (
+  <Stack direction="horizontal" spacing={8} width={300} height={50}>
+    <Box grow={1} />
```
Comment (yangwooseong) [inline .changeset/witty-pans-clean.md:25]:
```suggestion
    <Box grow={1} shrink={1} />
```

File: packages/bezier-react/src/components/ButtonGroup/ButtonGroup.types.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ interface ButtonGroupOptions {
 }
 
 export default interface ButtonGroupProps extends
-  BezierComponentProps,
```
Comment (yangwooseong) [inline packages/bezier-react/src/components/ButtonGroup/ButtonGroup.types.ts:17]: `BezierComponentProps` 를 deprecated 시켜도 좋을 것 같습니다

File: packages/bezier-react/src/components/Forms/FormGroup/FormGroup.types.ts (hunk 2 above)
Comment (sungik-choi) [reply packages/bezier-react/src/components/Forms/FormGroup/FormGroup.types.ts:11]: 이 부분 변경은 실수가 맞습니다..!

**Review by yangwooseong — APPROVED** [review]

**Pull request merged**